app/models/preserved_copy.rb

```diff
@@ -32,14 +32,31 @@ class PreservedCopy < ApplicationRecord
   validates :status, inclusion: { in: statuses.keys }
   validates :version, presence: true
 
-  scope :least_recent_version_audit, lambda { |last_checked_b4_date, storage_dir|
+  scope :by_endpoint_name, lambda { |endpoint_name|
+    joins(:endpoint).where(endpoints: { endpoint_name: endpoint_name })
+  }
+
+  scope :by_storage_location, lambda { |storage_dir|
+    joins(:endpoint).where(endpoints: { storage_location: storage_dir })
+  }
+
+  scope :least_recent_version_audit, lambda { |last_checked_b4_date|
     last_checked_b4_date = normalize_date(last_checked_b4_date)
-    joins(:endpoint)
-      .where(endpoints: { storage_location: storage_dir })
-      .where('last_version_audit IS NULL or last_version_audit < ?', last_checked_b4_date)
+    where('last_version_audit IS NULL or last_version_audit < ?', last_checked_b4_date)
       .order('last_version_audit IS NOT NULL, last_version_audit ASC')
   }
 
+  scope :fixity_check_expired, lambda {
+    joins(:preserved_object)
+      .joins(
+        'INNER JOIN preservation_policies'\
+        ' ON preservation_policies.id = preserved_objects.preservation_policy_id'\
+        ' AND (last_checksum_validation + (fixity_ttl * interval \'1 second\')) < CURRENT_TIMESTAMP'\
+        ' OR last_checksum_validation IS NULL'
+      )
+      .order('last_checksum_validation IS NOT NULL, last_checksum_validation ASC')
```

Comment:
~~shouldn't we be doing the null ones FIRST?~~  I see the tests confirm we are getting the right order.

Reply:
yeah, this is an unfortunately confusing sql thing.  oracle lets you just give a sort order and say `NULLS FIRST`.  other RDBMS software doesn't, though, as `NULLS FIRST` is an oracle-ism and not part of the vanilla sql standard.

since we've run into this counter-intuitive bit of sql before before, and since everyone (me included) first reads this `order by` as the opposite of what it really does, maybe a comment is in order?  see also https://github.com/sul-dlss/preservation_catalog/pull/519#discussion_r164555083

Reply:
yeah, a comment would be lovely.  Feel free to add one to analogous places elsewhere..